dashboard/app/models/authentication_option.rb

```diff
@@ -29,12 +29,32 @@ class AuthenticationOption < ApplicationRecord
   before_save :normalize_email, :hash_email, :remove_student_cleartext_email,
     :fill_authentication_id
 
+  OAUTH_CREDENTIAL_TYPES = [
+    CLEVER = 'clever',
+    FACEBOOK = 'facebook',
+    GOOGLE = 'google_oauth2',
+    POWERSCHOOL = 'powerschool',
+    QUIKCAMPS = 'lti_lti_prod_kids.qwikcamps.com',
+    THE_SCHOOL_PROJECT = 'the_school_project',
+    TWITTER = 'twitter',
```

Comment:
Providers we haven't tested and don't support migrating yet: Twitter, [on our list since at least September 2014](https://github.com/code-dot-org/code-dot-org/blame/092f72fdd84bb5ab6971aac55300a226b4fb4e98/dashboard/app/models/user.rb#L13). We have credentials in our production environment but no evidence yet that it's actually working, and not sure if any user accounts are set up for it.